src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -779,19 +778,33 @@ bool OutputStateMachineEngine::ActionOscDispatch(const wchar_t /*wch*/,
             TermTelemetry::Instance().Log(TermTelemetry::Codes::OSCWT);
             break;
         case OscActionCodes::SetColor:
-            success = _dispatch->SetColorTableEntry(tableIndex, color);
+            for (size_t i = 0; i < tableIndexes.size(); i++)
+            {
+                auto tableIndex = tableIndexes.at(i);
+                auto rgb = colors.at(i);
+                success = _dispatch->SetColorTableEntry(tableIndex, rgb);

```

Comment:
Ideally the `rgb` here is for sure valid. I'm not clear how `success` should be handled, though. Maybe this is better:

```cpp
success |= _dispatch->SetColorTableEntry(tableIndex, rgb);
```

Reply:
Given that a valid string we didn't handle sets `success` to false, we probably want this to be `success &= _dispatch->` so that we log the telemetry and trace in from `StateMachine::_ActionOscDispatch` if _any_ of the colour sets failed.

I'm not sure it makes sense to call `_pfnFlushToTerminal()` in the case of a string we understood, parsed, but failed to apply for some other reason, but the rest of the code here is doing that too.

Reply:
On the contrary I think we should consider the dispatch as a success as long as one of the colors is a success. Because the successful index will change the color in the table. Given that most escape sequences are generated by programs, invalid sequences shouldn’t really matter that much.

Reply:
I guess it depends on how you define "success". As I mentioned, the rest of this function seems to set success to false with even the vaguest hint of not-perfect handling. I was originally going to agree with `|=` when I wrote my earlier comment, but on looking at the context, it seems "success" here means "so totally handled that we should hide it from the rest of the system", while `!success` is "something didn't go perfectly, so log it".

For example "OSC code we don't recognise" is valid to consume-and-ignore (from what I could read, anyway), but still sets `success=false`.

Really, `success` by this point is no longer "success" but inverted "should we also pass this through to the TTY". See the comments in [`AdaptDispatch::SetColorTableEntry`](https://github.com/microsoft/terminal/blob/cb037f3953d1df36f7f62b62860ada1a6e307402/src/terminal/adapter/adaptDispatch.cpp#L2258-L2261) for where this is done even for sequences we _did_ handle fully.

Reply:
Given that `success` must be `true` to hit this point, `|=` can never change the value. And if I'm reading the behaviour of [`AdaptDispatch::SetColorTableEntry`](https://github.com/microsoft/terminal/blob/cb037f3953d1df36f7f62b62860ada1a6e307402/src/terminal/adapter/adaptDispatch.cpp#L2258-L2261) correctly, it relies on being able to make this `false` here?